### ISSUE TITLE
changing vjs-media just removes the player

### ISSUE THREAD

**Issue by rlaphoenix** [LonnyGomes/vjs-video#81]
$rootScope.vjsSources = {
        sources: [
          {
            src: "x.mp4",
            type: 'video/mp4'
          }
        ]
};

If I do the above, it just removes the player and leaves only the vjs-wrap div empty with nothing else on it. Using latest non release.

Edit:

I did some small investigating using an alert() to halt execution at points of code and it seems to be occuring around the $destroy emitter. Right after it gets destroyed, it stays EXACTLY like that, and doesnt do anything. I see that its meant to .append "compiledEl" but it doesnt append anything from what I can see.
CompiledEl does contain atleast 2 index, "0" and "context" both look similar.
why is compiledEl being given a $compile() thing after it gets appended? Even if I put that before it, it does nothing.

**Comment by rlaphoenix** [comment]
Ok dont have a fucking clue why, but if I leave the origin and compiledel vars etc completely alone (if I dont do shortcuts with em like parentContainer.append(origContent.clone());) and if I add the following after staleChild.remove(); must be after: 
                        parentContainer.append($compile(origContent)(scope.$parent));

It works? WHAT?
I didnt test Poster, but the video source is definitely changing and works too, I have no idea if events etc work properly though

**Comment by rlaphoenix** [comment]
I made a new Emit Event called vJsMediaToggleDone that emits to the scope after the append thing I did above so that I know when its appended and ready. I did a $scope.$on on it and then do a .addEventListener() to it for timeupdate and it seems to be added fine, but no events are being emitted from videojs's end after changing the sources like this.

Any ideas?

**Comment by rlaphoenix** [comment]
Nevermind, the events do work!
SO I guess I found the solution?!
My events issue is that it was targeting my player id for the original initial <video></video> tag, but that ends up being a div from the changes, so simply adding _html5_api at the end of the ID for example videoPlayer_html5_api will target the <video> tag, of course you could just get the first child of the videoPlayer ID just I find this way of doing _html5_api a bit faster and has less work needed.

**Comment by rlaphoenix** [comment]
And again, poster updates fine too :D

**Comment by ghost** [comment]
@PR4GM4 There is some tricky DOM stuff happening behinds the scenes. So all is working now?

**Comment by rlaphoenix** [comment]
Yeah once I add that string of text, from what I can see, the variable for compiledEl is essentially a one time use thing, so I dont see why it works when I leave that stuff alone (Cause the .Append does nothing). So when I do the extra append from directly origContent with $compile of parent scope, it actually appends and works as expected. Maybe you should look into this a bit?

**Comment by ghost** [comment]
@PR4GM4 yeah, I'm about to do a refactor things and will take a look

**Comment by rlaphoenix** [comment]
An update, seems the changes I made still cause one issue, if you change the source a 2nd time, it have staleChild return null or undefined, leaving 2 players in the source.

`Cannot read property 'remove' of undefined`
